crates/sui-graphql-rpc/src/raw_query.rs

```diff
@@ -179,6 +179,25 @@ macro_rules! or_filter {
     }};
 }
 
+/// Accepts two `Rawquery` instances and a third expression consisting of which columns to join on.
+#[macro_export]
+macro_rules! inner_join {
+    ($lhs:expr, $rhs:ident => ($rhs_query:expr, $alias:expr), using: [$using:expr $(, $more_using:expr)*]) => {{
+        use $crate::raw_query::RawQuery;
+
+        let (lhs_sql, mut binds) = $lhs.finish();
+        let (rhs_sql, rhs_binds) = $rhs_query.finish();
+
+        binds.extend(rhs_binds);
+
+        let aliased = format!("({}) AS {}", rhs_sql, $alias);
+        let using_clause = format!("USING ({})", stringify!($using $(, $more_using)*));
+        let sql = format!("{} INNER JOIN {} {}", lhs_sql, aliased, using_clause);
```

Comment:
nit: struggled to follow this a bit seeing as its all split up, how about:

```suggestion
        let sql = format!(
            "{lhs_sql} INNER JOIN ({rhs_sql}) AS {} USING({})", 
            $alias, 
            stringify!($using $(, $more_using)*),
        );
```

Also, note that it doesn't seem like you use `$rhs` at all? The intention was for that to be the alias in my original suggestion).